lib/matplotlib/axes/_axes.py

```diff
@@ -1931,37 +1931,63 @@ def step(self, x, y, *args, **kwargs):
         """
         Make a step plot.
 
+        Call signatures::
+
+            plot(x, y, [fmt], data=None, where='pre', **kwargs)
```

Comment:
`data` should probably go after kwargs as it is a keyword-only argument.

Reply:
But that might look odd, since the **kwargs form has to be at the end when one actually calls a function.   I would put `data` after `where`, since `data` is more generic, but I wouldn't put it after `**kwargs`

Reply:
That'd be a bit awkward. AFAIK, explicit arguments are not allowed after `**kwargs`. I can switch to the Python 3 syntax `plot(x, y, [fmt], *, data=None, where='pre', **kwargs)`.